bc_obps/registration/api/operation.py

```diff
@@ -44,83 +56,84 @@ def save_multiple_operators(multiple_operators_array: List[MultipleOperator], op
+
+        if operator.get("mo_mailing_address_same_as_physical"):
+            new_operator["mailing_address_id"] = physical_address.id
+        else:
+            mailing_address = Address.objects.create(
+                street_address=operator.get("mo_mailing_street_address"),
+                municipality=operator.get("mo_mailing_municipality"),
+                province=operator.get("mo_mailing_province"),
+                postal_code=operator.get("mo_mailing_postal_code"),
+            )
+            new_operator["mailing_address_id"] = mailing_address.id
 
         for field in operator:
             if field in multiple_operator_fields_mapping:
                 new_operator[multiple_operator_fields_mapping[field]] = operator[field]
 
         new_operator["business_structure"] = BusinessStructure.objects.get(name=operator["mo_business_structure"])
-
         # TODO: archive multiple operators in #361 that are not in the array anymore once #326 is done
 
         # check if there is a multiple_operator with that operation id and number
         # if there is, update it, if not, create it
-        existing_multiple_operator = MultipleOperator.objects.filter(
-            operation_id=operation.id, operator_index=idx + 1
-        ).first()
-        if existing_multiple_operator:
-            MultipleOperator.objects.filter(operation_id=operation.id, operator_index=idx + 1).update(**new_operator)
-            existing_multiple_operator.set_create_or_update(modifier=user)
-        else:
-            new_multiple_operator = MultipleOperator.objects.create(**new_operator)
-            new_multiple_operator.set_create_or_update(modifier=user)
+        multiple_operator, _ = MultipleOperator.objects.update_or_create(
+            operation_id=operation.id, operator_index=idx + 1, defaults={**new_operator}
+        )
+        multiple_operator.set_create_or_update(modifier=user)
 
 
 ##### GET #####
 
 
 @router.get("/operations", response={200: List[OperationOut], codes_4xx: Message})
+@authorize(AppRole.get_all_eligible_roles())
 def list_operations(request):
-    raise_401_if_role_not_authorized(request, ["industry_user", "industry_user_admin", "cas_admin", "cas_analyst"])
-    user = request.current_user
+    user: User = request.current_user
     # IRC users can see all operations except ones that are not registered yet
-    if user.app_role.role_name in ['cas_admin', 'cas_analyst']:
+    if user.is_irc_user():
         qs = Operation.objects.exclude(status__in=[Operation.Statuses.NOT_REGISTERED])
-        return qs
+        return 200, qs
     # Industry users can only see their companies' operations (if there's no user_operator or operator, then the user hasn't requested access to the operator)
-    user_operator = UserOperator.objects.filter(user_id=user.user_guid).first()
+    user_operator = UserOperator.objects.filter(user_id=user.user_guid).first()  # Why the first one??
```

Comment:
If we assume this is valid, we should constrain the UserOperator data model to prevent creating duplicate instances for the same user. Additionally, we need to consider cases where a user requests admin for multiple operators.

Reply:
Yes, I was working on something earlier, created a duplicate instance for the same user, and broke a lot. Do we have a separate ticket/plan for that? I don't see a change in this PR

Reply:
Once Dylan approves, we can add uniqueness to the UserOperator and safely assume each user has only one instance. Then we can update the code accordingly.